client/src/Components/SingularPost.js

```diff
@@ -18,6 +18,20 @@ export default function SingularPost(props) {
     setOpen(false);
   };
 
+  // useEffect(() => {
+  //   if (open) {
+  //     fetch("posts/" + user)
+  //       .then(res => res.json())
+  //       .then(res => {
+  //         if (res.error) {
+  //           console.log(res.error);
+  //         } else {
+  //           setUserPosts(res);
+  //         }
+  //       });
+  //   }
+  // }, [open]);
```

Comment:
You'd use the authFetch utility here, since this request requires the user to be authenticated

Reply:
Here you would fetch the comments and likes though, user posts would be fetched on the profile

Reply:
Also keep in mind that if the user opens it once, closes and open again, you don't want to fetch the same data 2x

Reply:
1) Appreciate the head's up on the authFetch!
2) I'm thinking a posts/{post_id}/comments route is the way to go.
3) Yep!  I'm hoping an if(open) block will solve that problem

Reply:
Also, if you are going to use authFetch, don't forget to protect your routes with the auth middleware on the back-end.

Reply:
Thanks Badr!  I'll likely just try to replicate what you have done elsewhere.